### ISSUE TITLE
Warning "References to the global RegExp object prevents optimization of regular expressions."

### ISSUE THREAD

**Issue by danielcompton** [pkpkpk/fress#7]
When compiling under `:advanced` (with Shadow CLJS), the following warning gets printed:

```
------ WARNING #2 -  -----------------------------------------------------------
 File: ~/.m2/repository/fress/fress/0.3.1/fress-0.3.1.jar!/fress/writer.cljs:452:5
--------------------------------------------------------------------------------
 449 |     (writeInt wtr length)
 450 |     (doseq [o arr] (writeObject wtr o))))
 451 | 
 452 | (def default-write-handlers
-----------^--------------------------------------------------------------------
 References to the global RegExp object prevents optimization of regular expressions.
--------------------------------------------------------------------------------
 453 |   {js/Number writeNumber
 454 |    js/String writeString
 455 |    js/Boolean writeBoolean
 456 |    js/Array writeList
```

https://github.com/plumatic/schema/pull/165 shows one way that this was previously solved, though I'm not sure if it's directly applicable here.

**Comment by pkpkpk** [comment]
We need the constructor dynamically here so i think an inline suppress hint to closure is probably whats needed. Ill look into it

**Comment by danielcompton** [comment]
@pkpkpk thanks! One thing I was unsure about was if the suppression still allows optimisation, or if it just disables the warning. Hopefully the former.

**Comment by pkpkpk** [comment]
Im pretty sure tags are only local in scope and global effects have to be set in compiler options, so it shouldnt affect other code. Also FWIW closure [check-regexp only parses regexps for syntax/flag validity](https://github.com/google/closure-compiler/blob/7ff6e25843097791416d1544c88dc8711f6be64a/src/com/google/javascript/jscomp/CheckRegExp.java#L105). Things like hoisting out calls are treated like any other function and are not subject to suppression. Since we dont have patterns to be statically checked here, there's nothing to lose

That said I think the top level map is unnecessarily heavy and simple table should suffice. I will close this when i swap it over

**Comment by pkpkpk** [comment]
should be fixed in `"0.3.3"`

**Comment by danielcompton** [comment]
Thank you, looks good! https://github.com/pkpkpk/fress/commit/67923b75e7326787b1fd54970530cdb677dd3a7d
